### PR TITLE
Add /setgame and /settitle

### PR DESCRIPTION
**Pull request by Mm2PL**
Pull request checklist:

- [x] `CHANGELOG.md` was updated, if applicable

# Description

Cherrypicked from Mm2PL/dankerino#8 and Mm2PL/dankerino#9

### CODE REVIEW

File: src/controllers/commands/CommandController.cpp [hunk 1]
```diff
@@ -643,7 +643,85 @@ void CommandController::initialize(Settings &, Paths &paths)
...
+                "Unable to set title of non-Twitch channel."));
+        }
+        return "";
+    });
+    this->registerCommand("/setgame", [](const QStringList &words,
+                                         ChannelPtr channel) {
+        if (words.size() < 2)
+        {
+            channel->addMessage(
+                makeSystemMessage("This command comes from Dankerino patches. "
```
Comment (pajlada) [inline src/controllers/commands/CommandController.cpp:684]: Fix this message to not include Dankerino patches messaging

File: src/providers/twitch/api/Helix.cpp [hunk 2]
```diff
@@ -578,6 +578,48 @@ void Helix::unblockUser(QString targetUserId,
         .execute();
 }
 
+void Helix::updateChannel(QString broadcasterId, QString gameId,
```
Comment (pajlada) [inline src/providers/twitch/api/Helix.cpp:581]: Could you update the `src/providers/twitch/api/README.md` with the information about this request, including which scope is required?

Comment (pajlada) [reply src/providers/twitch/api/Helix.cpp:581]: preferably it should go right below the `Get Channel` entry on line 99

File: src/controllers/commands/CommandController.cpp [hunk 3]
```diff
@@ -643,7 +643,85 @@ void CommandController::initialize(Settings &, Paths &paths)
...
+                    }
+                    else  // 0 or 1 games
+                    {
+                        auto status = twitchChannel->accessStreamStatus();
+                        getHelix()->updateChannel(
+                            twitchChannel->roomId(), games.at(0).id, "", "",
+                            [channel](NetworkResult dank) {
+                                channel->addMessage(
+                                    makeSystemMessage(QString("Updated game.")
+                                                          .arg(dank.status())));
```
Comment (pajlada) [inline src/controllers/commands/CommandController.cpp:706]: unused argument

File: src/controllers/commands/CommandController.cpp [hunk 4]
```diff
@@ -643,7 +643,85 @@ void CommandController::initialize(Settings &, Paths &paths)
...
+            return "";
+        }
+        if (auto twitchChannel = dynamic_cast<TwitchChannel *>(channel.get()))
+        {
+            auto status = twitchChannel->accessStreamStatus();
+            getHelix()->updateChannel(
+                twitchChannel->roomId(), "", "", words.mid(1).join(" "),
+                [channel](NetworkResult dank) {
+                    channel->addMessage(makeSystemMessage(
+                        QString("Updated title.").arg(dank.status())));
```
Comment (pajlada) [inline src/controllers/commands/CommandController.cpp:664]: unused argument

File: src/controllers/commands/CommandController.cpp [hunk 5]
```diff
@@ -643,7 +643,85 @@ void CommandController::initialize(Settings &, Paths &paths)
...
+                },
+                [channel] {
+                    channel->addMessage(
```
Comment (pajlada) [inline src/controllers/commands/CommandController.cpp:667]: Is it possible to forward the errors here so we can be a bit more intelligently informing the user if they're missing the scope or if the network connectivity is shit or what not? not a required change, just spitballing

File: src/controllers/commands/CommandController.cpp [hunk 6]
```diff
@@ -643,7 +643,85 @@ void CommandController::initialize(Settings &, Paths &paths)
...
+                            makeSystemMessage("Game not found."));
+                    }
+                    else  // 0 or 1 games
+                    {
+                        auto status = twitchChannel->accessStreamStatus();
+                        getHelix()->updateChannel(
+                            twitchChannel->roomId(), games.at(0).id, "", "",
+                            [channel](NetworkResult dank) {
+                                channel->addMessage(
+                                    makeSystemMessage(QString("Updated game.")
```
Comment (pajlada) [inline src/controllers/commands/CommandController.cpp:705]: Could we add messaging about which game has been set? if it's available in games.at(0).name

**Review by pajlada — CHANGES_REQUESTED** [review]

File: src/controllers/commands/CommandController.cpp (hunk 6 above)
Comment (Mm2PL) [reply src/controllers/commands/CommandController.cpp]: Added game name in 516b17ae

File: src/controllers/commands/CommandController.cpp (hunk 4 above)
Comment (Mm2PL) [reply src/controllers/commands/CommandController.cpp]: Replaced with useful argument in 516b17ae

File: src/controllers/commands/CommandController.cpp (hunk 3 above)
Comment (Mm2PL) [reply src/controllers/commands/CommandController.cpp]: Replaced with useful argument in 516b17ae

File: src/controllers/commands/CommandController.cpp (hunk 1 above)
Comment (Mm2PL) [reply src/controllers/commands/CommandController.cpp]: 516b17ae

File: src/providers/twitch/api/README.md [hunk 7]
```diff
@@ -104,6 +104,14 @@ URL: https://dev.twitch.tv/docs/api/reference#get-channel-information
   Used in:
   - `TwitchChannel` to refresh stream title
 
+### Update Channel
+URL: https://dev.twitch.tv/docs/api/reference#modify-channel-information
```
Comment (pajlada) [inline src/providers/twitch/api/README.md:109]: Please add the required scope here, see the "Create Clip" entry for reference

**Review by pajlada — CHANGES_REQUESTED** [review]

File: src/controllers/commands/CommandController.cpp (hunk 5 above)
Comment (Mm2PL) [reply src/controllers/commands/CommandController.cpp:668]: This would require changing Helix error handling which is out of scope of this PR.

**Review by pajlada — APPROVED** [review]
Looks good to me! Thank you for your contribution :+1:

**Pull request merged**